consensus/src/payload_manager.rs

```diff
@@ -586,14 +596,43 @@ async fn process_payload_helper<T: TDataInfo>(
     batch_reader: Arc<dyn BatchReader>,
     block: &Block,
     ordered_authors: &[PeerId],
+    additional_peers_to_request: Option<&BitVec>,
 ) -> ExecutorResult<Vec<SignedTransaction>> {
-    let (iteration, fut) = {
+    let (iteration, fut, existing_responders) = {
         let data_fut_guard = data_ptr.data_fut.lock();
         let data_fut = data_fut_guard.as_ref().expect("must be initialized");
-        (data_fut.iteration, data_fut.fut.clone())
+        (
+            data_fut.iteration,
+            data_fut.fut.clone(),
+            data_fut.responders.clone(),
+        )
+    };
+
+    let mut signers = Vec::new();
+    if let Some(peers) = additional_peers_to_request {
+        for i in peers.iter_ones() {
+            if let Some(author) = ordered_authors.get(i) {
+                signers.push(*author);
+            }
+        }
+    }
+
+    let result = {
+        // If the future is completed then use the result.
+        if let Some(result) = fut.clone().now_or_never() {
```

Comment:
is it worth having this early check? just appending the responders doesn't seem too bad?

Reply:
yeah, makes sense. we compute it anyhow. I will remove it.